open-api/src/testFixtures/java/org/apache/iceberg/rest/RESTCatalogServer.java

```diff
@@ -0,0 +1,123 @@
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RESTCatalogServer {
+  private static final Logger LOG = LoggerFactory.getLogger(RESTCatalogServer.class);
+
+  static final String REST_PORT = "rest.port";
+  static final int REST_PORT_DEFAULT = 8181;
+
+  private Server httpServer;
+
+  RESTCatalogServer() {}
+
+  static class CatalogContext {
+    private final Catalog catalog;
+    private final Map<String, String> configuration;
+
+    CatalogContext(Catalog catalog, Map<String, String> configuration) {
+      this.catalog = catalog;
+      this.configuration = configuration;
+    }
+
+    public Catalog catalog() {
+      return catalog;
+    }
+
+    public Map<String, String> configuration() {
+      return configuration;
+    }
+  }
+
+  private CatalogContext initializeBackendCatalog() throws IOException {
+    // Translate environment variables to catalog properties
+    Map<String, String> catalogProperties = RCKUtils.environmentCatalogConfig();
+
+    // Fallback to a JDBCCatalog impl if one is not set
+    catalogProperties.putIfAbsent(CatalogProperties.CATALOG_IMPL, JdbcCatalog.class.getName());
+    catalogProperties.putIfAbsent(CatalogProperties.URI, "jdbc:sqlite::memory:");
+    catalogProperties.putIfAbsent("jdbc.schema-version", "V1");
+
+    // Configure a default location if one is not specified
+    String warehouseLocation = catalogProperties.get(CatalogProperties.WAREHOUSE_LOCATION);
+
+    if (warehouseLocation == null) {
+      File tmp = java.nio.file.Files.createTempDirectory("iceberg_warehouse").toFile();
+      tmp.deleteOnExit();
+      warehouseLocation = tmp.toPath().resolve("iceberg_data").toFile().getAbsolutePath();
+      catalogProperties.put(CatalogProperties.WAREHOUSE_LOCATION, warehouseLocation);
+
+      LOG.info("No warehouse location set. Defaulting to temp location: {}", warehouseLocation);
+    }
+
+    LOG.info("Creating catalog with properties: {}", catalogProperties);
+    return new CatalogContext(
+        CatalogUtil.buildIcebergCatalog("rest_backend", catalogProperties, new Configuration()),
+        catalogProperties);
+  }
+
+  public void start(boolean join) throws Exception {
+    CatalogContext catalogContext = initializeBackendCatalog();
+
+    RESTCatalogAdapter adapter = new RESTServerCatalogAdapter(catalogContext);
+    RESTCatalogServlet servlet = new RESTCatalogServlet(adapter);
+
+    ServletContextHandler context = new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
+    ServletHolder servletHolder = new ServletHolder(servlet);
+    context.addServlet(servletHolder, "/*");
+    context.insertHandler(new GzipHandler());
+
+    this.httpServer =
+        new Server(
+            PropertyUtil.propertyAsInt(catalogContext.configuration, REST_PORT, REST_PORT_DEFAULT));
+    httpServer.setHandler(context);
+    httpServer.start();
+
+    if(join) {
```

Comment:
nit: looks like formatting is off here. I actually wonder whether spotless is being applied to this project

Reply:
I double-checked and it indeed looks like spotless is only applied to src + test but not to the code in testFixtures, so we might want to look into that

Reply:
I'll follow up with a PR for this.

Reply:
@nastra you were right, [PR here](https://github.com/apache/iceberg/pull/11024)